include/seastar/core/seastar.hh

```diff
@@ -384,4 +384,29 @@ future<uint64_t> fs_avail(std::string_view name) noexcept;
 future<uint64_t> fs_free(std::string_view name) noexcept;
 /// @}
 
+/// \defgroup subprocess-module Subprocess Management
+
+/// Spawn a subprocess
+///
+/// \param pathname the path to the executable
+/// \param argv parameters psassed in the command line
+/// \param env optional environment variables
+///
+/// \return a tuple of pid, stdin, stdout and stderr fd
+/// \note
+/// the subprocess is spawned with \c posix_spawn()
+/// system call, so the pathname should be relative or
+/// absolute path of the executable.
+future<std::tuple<pid_t, int, int, int>>
+spawn(std::string_view pathname,
+      std::vector<sstring> argv,
+      std::vector<sstring> env = {});
```

Comment:
Idea: change the signature to `span(std::filesystem::path path, spawn_parameters params)`. This way when we later add more parameters from posix_spawn() we don't need to add more and more overloads.

Also, shouldn't it return a seastar::process?

Reply:
love it! actually, i wanted to add
```c++
bool stdin;
bool stdout;
bool stderr;
```
parameters allowing user to enable/disable writing/reading to/from the spawned process. but on the other hand, i don't want to have an overcomplicated API like Python's `subprocess.Popen()` constructor. this setting is a nice compromise, i think.